### ISSUE TITLE
DNS Discovery exceptions during startup

### ISSUE THREAD

**Issue by gezero** [hyperledger/besu#3044]
### Description
When starting besu with the DNSResolver fix in I can see several exceptions in the logs of the following form:
```
2021-11-11 13:16:07.530+00:00 | Timer-0 | WARN  | DNSResolver | I/O exception contacting remote DNS server /8.8.8.8:53
java.io.EOFException
        at org.xbill.DNS.NioTcpClient$ChannelState.processRead(NioTcpClient.java:169)
        at org.xbill.DNS.NioTcpClient$ChannelState.processReadyKey(NioTcpClient.java:126)
        at org.xbill.DNS.NioClient.processReadyKeys(NioClient.java:143)
        at org.xbill.DNS.NioClient.runSelector(NioClient.java:118)
        at java.base/java.lang.Thread.run(Thread.java:829)
```
The exception seems to be raised when the `tuweni-dns-discovery-2.0.0.jar` library is trying to walk the merkle tree of the [discovery protocol](https://eips.ethereum.org/EIPS/eip-1459). My suspicion is that the algorithm is producing too many DNS requests and the DNS server returns empty response from time to time. The warning is printed in the `DNSResolver::resolveRecordRaw` method.  I was able to recover domain names that produced the error on my machine and they seem to be random in nature.  For instance when running against the goerli Testnet I saw the following domains to fail:
```
run1:
GLZ7LY2CB2AOW2HK6QPCDJZSI4.all.goerli.ethdisco.net
OJXXBIYTK4VAS7YSTPACLAYRHY.all.goerli.ethdisco.net
RY6UILVC5GIJHFLBFKVESYZQ7E.all.goerli.ethdisco.net
KY2MQGH456TGZ63EK3QTS44NFI.all.goerli.ethdisco.net

run2:
GLZ7LY2CB2AOW2HK6QPCDJZSI4.all.goerli.ethdisco.net
LCUHVFB4LZ7C4SATYHEZUGBTLY.all.goerli.ethdisco.net
UXGZS3OCFEWXE66A235P3COXJI.all.goerli.ethdisco.net
S3ELF47KWAXNGC5RX7BTFP.all.goerli.ethdisco.net
3LP6WAS5OGPDEN634PUBTLXY7M.all.goerli.ethdisco.net

run3:
YELXTO5RLLVTPTH4HQFNKASOGU.all.goerli.ethdisco.net
LCUHVFB4LZ7C4SATYHEZUGBTLY.all.goerli.ethdisco.net
UXGZS3OCFEWXE66A235P3COXJI.all.goerli.ethdisco.net
S3ELF47KWAXNGC5RX7BTFP.all.goerli.ethdisco.net
3LP6WAS5OGPDEN634PUBTLXY7M.all.goerli.ethdisco.net
```

I checked manually and at the time I was testing it the entries was possible to retrieve without problem when requesting in isolation. For instance, consider:
```
 $ dig +tcp S3ELF47KWAXNGC5RX7BTFP.all.goerli.ethdisco.net TXT

; <<>> DiG 9.16.1-Ubuntu <<>> +tcp S3ELF47KWAXNGC5RX7BTFP.all.goerli.ethdisco.net TXT
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NXDOMAIN, id: 17237
;; flags: qr rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 1, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 512
;; QUESTION SECTION:
;S3ELF47KWAXNGC5RX7BTFP.all.goerli.ethdisco.net.        IN TXT

;; AUTHORITY SECTION:
ethdisco.net.           900     IN      SOA     ns-1441.awsdns-52.org. awsdns-hostmaster.amazon.com. 1 7200 900 1209600 86400

;; Query time: 10 msec
;; SERVER: 8.8.8.8#53(8.8.8.8)
;; WHEN: Thu Nov 11 17:21:01 GMT 2021
;; MSG SIZE  rcvd: 160
```

When looking at the code, it seems that the algorithm implementation:
* does not rate limit calls
* does not handle the exception sensibly - ignoring the subtree of the merkle tree where the failure happened
* swallows the domain name and prints a warning without even telling what domain name failed

### Acceptance Criteria
* Not having exceptions in the logs
* properly processing the whole merkle tree 
### Steps to Reproduce (Bug)
1. start the recent build of besu and check the log

**Frequency:** [What percentage of the time does it occur?]
Most of the DNS request pass, just some of them fail. In isolation, when stopped in between requests it seems all the requests succeeded

### Versions (Add all that apply)
* Java version: [`openjdk version "11.0.11" 2021-04-20`]
* OS Name & Version: [`Ubuntu 20.04 LTS as WSL under Windows 10]

**Comment by MadelineMurray** [comment]
hi @gezero - looks like you're looking at this one.  Can you confirm?  I've assigned to you in the meantime.

**Comment by gezero** [comment]
Confirming, thank you.

**Comment by macfarla** [comment]
@gezero Tuweni getting close to a next release - RC https://github.com/apache/incubator-tuweni/releases/tag/v2.1.0-incubating-rc

**Comment by gezero** [comment]
It seems to me that #3281 indeed resolved this particular problem

**Comment by tiagocmachado** [comment]
Seeing this again on besu `23.1.1`.

```
Mar 08 07:32:17 mainnet-8 besu[425459]: 2023-03-08 07:32:17.475+01:00 | Timer-0 | WARN  | DNSResolver | I/O exception contacting remote DNS server when resolving 4WWV2BBZ6YP2ZFZ4KTMCYRPO6E.all.mainnet.ethdisco.net
Mar 08 07:32:17 mainnet-8 besu[425459]: java.io.IOException: Timed out while trying to resolve 4WWV2BBZ6YP2ZFZ4KTMCYRPO6E.all.mainnet.ethdisco.net./TXT, id=22933
Mar 08 07:32:17 mainnet-8 besu[425459]:         at org.xbill.DNS.Resolver.send(Resolver.java:170)
Mar 08 07:32:17 mainnet-8 besu[425459]:         at org.apache.tuweni.discovery.DNSResolver.resolveRecordRaw(DNSResolver.kt:155)
Mar 08 07:32:17 mainnet-8 besu[425459]:         at org.apache.tuweni.discovery.DNSResolver.resolveRecord(DNSResolver.kt:73)
Mar 08 07:32:17 mainnet-8 besu[425459]:         at org.apache.tuweni.discovery.DNSResolver.internalVisit(DNSResolver.kt:175)
Mar 08 07:32:17 mainnet-8 besu[425459]:         at org.apache.tuweni.discovery.DNSResolver.internalVisit(DNSResolver.kt:184)
Mar 08 07:32:17 mainnet-8 besu[425459]:         at org.apache.tuweni.discovery.DNSResolver.internalVisit(DNSResolver.kt:184)
Mar 08 07:32:17 mainnet-8 besu[425459]:         at org.apache.tuweni.discovery.DNSResolver.internalVisit(DNSResolver.kt:184)
Mar 08 07:32:17 mainnet-8 besu[425459]:         at org.apache.tuweni.discovery.DNSResolver.internalVisit(DNSResolver.kt:184)
Mar 08 07:32:17 mainnet-8 besu[425459]:         at org.apache.tuweni.discovery.DNSResolver.visitTree(DNSResolver.kt:135)
Mar 08 07:32:17 mainnet-8 besu[425459]:         at org.apache.tuweni.discovery.DNSResolver.visitTree(DNSResolver.kt:111)
Mar 08 07:32:17 mainnet-8 besu[425459]:         at org.apache.tuweni.discovery.DNSResolver.collectAll(DNSResolver.kt:95)
Mar 08 07:32:17 mainnet-8 besu[425459]:         at org.apache.tuweni.discovery.DNSTimerTask.run(DNSDaemon.kt:89)
Mar 08 07:32:17 mainnet-8 besu[425459]:         at java.base/java.util.TimerThread.mainLoop(Timer.java:566)
Mar 08 07:32:17 mainnet-8 besu[425459]:         at java.base/java.util.TimerThread.run(Timer.java:516)
Mar 08 07:32:24 mainnet-8 besu[425459]: 2023-03-08 07:32:24.896+01:00 | vert.x-worker-thread-0 | INFO  | AbstractEngineNewPayload | Imported #16,781,777 / 132 tx / base fee 21.00 gwei / 11,970,196 (39.9%) gas / (0x0ac019fa8047d3b0b9742256fece901d546f0af85669da2ea488119814cd33bc) in 0.463s. Peers: 8
Mar 08 07:32:37 mainnet-8 besu[425459]: 2023-03-08 07:32:37.454+01:00 | vert.x-worker-thread-0 | INFO  | AbstractEngineNewPayload | Imported #16,781,778 / 135 tx / base fee 20.47 gwei / 16,104,883 (53.7%) gas / (0xd79e81fce80499ee9d7eae7235f173673875df4089451a464c2bdd1616e8083e) in 0.606s. Peers: 13
Mar 08 07:32:42 mainnet-8 besu[425459]: 2023-03-08 07:32:42.487+01:00 | Timer-0 | WARN  | DNSResolver | I/O exception contacting remote DNS server when resolving OMVFASJSWU4ZBFCJH3V5ZY.all.mainnet.ethdisco.net
Mar 08 07:32:42 mainnet-8 besu[425459]: java.io.IOException: Timed out while trying to resolve OMVFASJSWU4ZBFCJH3V5ZY.all.mainnet.ethdisco.net./TXT, id=61145
Mar 08 07:32:42 mainnet-8 besu[425459]:         at org.xbill.DNS.Resolver.send(Resolver.java:170)
Mar 08 07:32:42 mainnet-8 besu[425459]:         at org.apache.tuweni.discovery.DNSResolver.resolveRecordRaw(DNSResolver.kt:155)
Mar 08 07:32:42 mainnet-8 besu[425459]:         at org.apache.tuweni.discovery.DNSResolver.resolveRecord(DNSResolver.kt:73)
Mar 08 07:32:42 mainnet-8 besu[425459]:         at org.apache.tuweni.discovery.DNSResolver.internalVisit(DNSResolver.kt:175)
Mar 08 07:32:42 mainnet-8 besu[425459]:         at org.apache.tuweni.discovery.DNSResolver.internalVisit(DNSResolver.kt:184)
Mar 08 07:32:42 mainnet-8 besu[425459]:         at org.apache.tuweni.discovery.DNSResolver.internalVisit(DNSResolver.kt:184)
Mar 08 07:32:42 mainnet-8 besu[425459]:         at org.apache.tuweni.discovery.DNSResolver.internalVisit(DNSResolver.kt:184)
Mar 08 07:32:42 mainnet-8 besu[425459]:         at org.apache.tuweni.discovery.DNSResolver.internalVisit(DNSResolver.kt:184)
Mar 08 07:32:42 mainnet-8 besu[425459]:         at org.apache.tuweni.discovery.DNSResolver.visitTree(DNSResolver.kt:135)
Mar 08 07:32:42 mainnet-8 besu[425459]:         at org.apache.tuweni.discovery.DNSResolver.visitTree(DNSResolver.kt:111)
Mar 08 07:32:42 mainnet-8 besu[425459]:         at org.apache.tuweni.discovery.DNSResolver.collectAll(DNSResolver.kt:95)
Mar 08 07:32:42 mainnet-8 besu[425459]:         at org.apache.tuweni.discovery.DNSTimerTask.run(DNSDaemon.kt:89)
Mar 08 07:32:42 mainnet-8 besu[425459]:         at java.base/java.util.TimerThread.mainLoop(Timer.java:566)
Mar 08 07:32:42 mainnet-8 besu[425459]:         at java.base/java.util.TimerThread.run(Timer.java:516)
Mar 08 07:32:47 mainnet-8 besu[425459]: 2023-03-08 07:32:47.503+01:00 | vert.x-worker-thread-0 | INFO  | AbstractEngineNewPayload | Imported #16,781,779 / 11 tx / base fee 20.66 gwei / 685,960 (2.3%) gas / (0xa31ce60c5e135898c9085cf5f6b89fe650677112d7fb20c56a1db14679e7c68e) in 0.069s. Peers: 9```

**Comment by Xexr** [comment]
Also seeing this same error on 23.1.1

**Comment by MrNeth** [comment]
Any fix for this im seeing this on v23.10.1
